### PR TITLE
Feature/consolidation yggdrasil artifacts

### PR DESCRIPTION
**Pull request by V4L3**

### CODE REVIEW

File: lib/src/main/java/org/hyperagents/jacamo/artifacts/yggdrasil/NotificationServerArtifact.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+                }
+            }
+
+            if (artifactIRI == null) {
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                response.setContentType("text/plain");
+                response.getWriter()
+                        .println("Link headers are missing! See the W3C WebSub Recommendation for details.");
+            } else {
+                /**
```
Comment (andreiciortea) [inline lib/src/main/java/org/hyperagents/jacamo/artifacts/yggdrasil/NotificationServerArtifact.java]: We should clean up this part

File: lib/src/main/java/org/hyperagents/jacamo/artifacts/yggdrasil/NotificationServerArtifact.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+
+import cartago.Artifact;
+import cartago.ArtifactId;
+import cartago.INTERNAL_OPERATION;
+import cartago.OPERATION;
+
+/**
```
Comment (andreiciortea) [inline lib/src/main/java/org/hyperagents/jacamo/artifacts/yggdrasil/NotificationServerArtifact.java:28]: We should document all operations exposed to agents

File: lib/src/main/java/org/hyperagents/jacamo/artifacts/yggdrasil/Notification.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package org.hyperagents.jacamo.artifacts.yggdrasil;
+
+public class Notification {
```
Comment (andreiciortea) [inline lib/src/main/java/org/hyperagents/jacamo/artifacts/yggdrasil/Notification.java:3]: We should document this class as well

File: lib/src/main/java/org/hyperagents/jacamo/artifacts/yggdrasil/YggdrasilThingArtifact.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        if (this.hasObsPropertyByTemplate(functor, (Object[]) params)) {
+            this.updateObsProperty(functor, (Object[]) params);
+        } else {
+            this.defineObsProperty(functor, (Object[]) params);
+        }
+    }
+
+    /*
+     * Registers for WebSub to an Yggdrasil node. This is not a generic
```
Comment (andreiciortea) [inline lib/src/main/java/org/hyperagents/jacamo/artifacts/yggdrasil/YggdrasilThingArtifact.java]: Ideally, we should improve the implementation such that it is no longer specific to Yggdrasil — but works for any ThingArtifact using WebSub.

**Review by andreiciortea — CHANGES_REQUESTED** [review]
Many thanks, @V4L3!

**Comment by V4L3** [comment]
@andreiciortea , thanks for the feedback.
I have made some adjustments:
- Added documentation
- Renamed YggdrasilThingArtifact to WebSubThingArtifact and moved it to the `wot` package
- Generalized the WebSub implementation to handle multiple cases: 2 Link headers, 1 Link header, Link tags in HTML
- Moved HMAS class to its own package
- Some Clean up
- Added an Example and adjusted the ReadMe.md

**Review by andreiciortea — APPROVED** [review]

**Review by andreiciortea — APPROVED** [review]
Great, thanks @V4L3 !

**Pull request merged**